src/pybind/mgr/cephadm/module.py

```diff
@@ -1318,6 +1313,11 @@ def _check_valid_addr(self, host: str, addr: str) -> str:
 > ssh -F ssh_config -i ~/cephadm_private_key {self.ssh_user}@{addr}'''
             raise OrchestratorError(msg)
 
+        if not addr or host == addr:  # In case we had to deduce the IP ourselves
+            if ipaddress.ip_address(ip_addr).is_loopback:
+                raise OrchestratorError(
+                    f'Cannot add host {host}: IP {ip_addr} is a loopback address')
```

Comment:
I feel like if a user runs something like "ceph orch add host1" and gets back a message saying some ip they never provided is a loopback address they'll get confused. I think we actually don't need to mention loopback addresses at all and can just tell the user we can't figure out the ip and they need to provide one. Something like

```suggestion
                    f'Cephadm failed to deduce a valid ip address for host {host}. Please explicity provide an address to add this host (ceph orch host add {host} --addr <ip-addr>')
```